src/Adapter/Product/AdminProductDataUpdater.php

```diff
@@ -169,6 +171,11 @@ public function duplicateProduct($productId, $namePattern = 'copy of %s')
             throw new \Exception('AdminProductDataUpdater->duplicateProduct() received an unknown ID.', 5005);
         }
 
+        if ((($product->validateFields(false, true)) !== true)
+            || (($product->validateFieldsLang(false, true)) !== true)) {
+            throw new UpdateProductException('Cannot duplicate many requested products', 5004);
```

Comment:
cc @LouiseBonnard

Reply:
Should we mention the error is caused by a too long product description?

Reply:
@LouiseBonnard, Do you mean specifically test description_short instead of all fields?
```suggestion
        foreach (Language::getIDs(false) as $id_lang) {
            if ($product->validateField('description_short', $product->description_short[$id_lang], $id_lang) !== true) {
                throw new UpdateProductException(
                    sprintf(
                        'Cannot duplicate this product because the "Summary" field for the language "%s" is too long',
                        Language::getIsoById($id_lang)
                    )
                );                                        
            }
        }
```

Reply:
@LouiseBonnard, do you mean specifically validate the description_short field instead of all fields? Maybe something like: 'Cannot duplicate this product because the "Summary" field for the language "%s" is too long'?

Reply:
@LouiseBonnard the issue can be a field too long, but something else too.

Il we want to display a reason, we could retrieve the result of `
validateFields` or `
validateFieldsLang` instead.

Reply:
Alright, I thought it was only because of the long description... thanks for the suggestion, @Quetzacoalt91, how could we display this wording?

Reply:
We could a "global" message saying we cannot duplicate the product (as initially suggested), and concatenate the returned validation message (already translated).

Reply:
@Quetzacoalt91, I have made some changes. Is that what you were talking about?